src/core/core.js

```diff
@@ -220,12 +233,25 @@ export default class Core {
    * @param {string} verticalKey The vertical key to use in the search
    */
   verticalPage (verticalKey) {
-    this.verticalSearch(verticalKey, { useFacets: true }, {
+    this.verticalSearch(verticalKey, { useFacets: true, setQueryParams: true }, {
       id: this.globalStorage.getState(StorageKeys.QUERY_ID)
     });
   }
 
-  search (queryString, urls) {
+  search (queryString, urls, options = {}) {
+    const { setQueryParams } = options;
+    const context = this.globalStorage.getState(StorageKeys.API_CONTEXT);
+    const referrerPageUrl = this.globalStorage.getState(StorageKeys.REFERRER_PAGE_URL);
+
+    if (setQueryParams) {
+      if (context) {
+        this.persistentStorage.set(StorageKeys.API_CONTEXT, context);
```

Comment:
do we need to set persistentStorage here for api context? Can we not do it in core.setContext?

Reply:
If we set persistentStorage in setContext, then the context will be applied as soon as `ANSWERS.setContext` is called, but the PM would like it to be applied only right before a search. That is, if we are on a redirectUrl searchbar and we use ANSWERS.setContext, it will not affect the client's URL, and will only show up on the experience after the search.